src/domain/event/__tests__/CreateEventPage.test.tsx

```diff
@@ -0,0 +1,406 @@
+  {
+    request: {
+      query: MyProfileDocument,
+    },
+    result: {
+      ...profileResponse,
+    },
+  },
+  {
+    request: {
+      query: UploadSingleImageDocument,
+      variables: {
+        image: {
+          name: '',
+          image: imageFile,
+        },
+      },
+    },
+    result: imageMutationResponse,
+  },
+  {
+    request: {
+      query: ImageDocument,
+      variables: {
+        id: '48584',
+      },
+    },
+    result: imageResponse,
+  },
+  {
+    request: {
+      query: KeywordsDocument,
+      skip: false,
+      variables: {
+        pageSize: AUTOSUGGEST_OPTIONS_AMOUNT,
+        text: 'perheet',
+      },
+    },
+    result: keywordsResponse,
+  },
+  {
+    request: {
+      query: PlacesDocument,
+      skip: false,
+      variables: {
+        dataSource: 'tprek',
+        pageSize: AUTOSUGGEST_OPTIONS_AMOUNT,
+        text: 'Sellon',
+      },
+    },
+    result: placesResponse,
+  },
+  {
+    request: {
+      query: PlaceDocument,
+      skip: false,
+      variables: {
+        id: 'tprek:15417',
+      },
+    },
+    result: placeResponse,
+  },
+  {
+    request: {
+      query: CreateEventDocument,
+      variables: createEventVariables,
+    },
+    result: addEventResponse,
+  },
+];
+
+test('is accessible', async () => {
+  const { container } = render(
+    <Provider store={store}>
+      <MockedProvider mocks={mocks}>
+        <CreateEventPage />
+      </MockedProvider>
+    </Provider>
+  );
+
+  await waitFor(() => {
+    expect(
+      screen.queryByText('Kulttuuri- ja vapaa-aikalautakunnan kulttuurijaosto')
+    ).toBeInTheDocument();
+  });
+
+  const result = await axe(container);
+  expect(result).toHaveNoViolations();
+});
+
+test.only('form works correctly when edited', async () => {
+  advanceTo(new Date(2020, 7, 8));
+  const pushMock = jest.fn();
+  jest.spyOn(Router, 'useHistory').mockReturnValue({
+    push: pushMock,
+  } as any);
+  const { container } = render(
+    <Provider store={store}>
+      <MockedProvider mocks={mocks}>
+        <CreateEventPage />
+      </MockedProvider>
+    </Provider>
+  );
+
+  Modal.setAppElement(container);
+
+  await waitFor(() => {
+    expect(
+      screen.queryByText('Kulttuuri- ja vapaa-aikalautakunnan kulttuurijaosto')
+    ).toBeInTheDocument();
+  });
+
+  userEvent.type(screen.getByLabelText('Tapahtuman nimi'), eventFormData.name);
+
+  expect(screen.getByTestId('event-form')).toHaveFormValues({
+    name: eventFormData.name,
+  });
+
+  // should open modal when trying to change event language
+  userEvent.click(screen.getByRole('button', { name: 'Ruotsi', hidden: true }));
+  expect(screen.getByRole('dialog')).toHaveTextContent(/vaihda kieli/i);
+
+  const modal = within(screen.getByRole('dialog', { hidden: true }));
+
+  const cancelButton = modal.getByRole('button', {
+    name: 'Peruuta',
+    hidden: true,
+  });
+  userEvent.click(cancelButton);
+
+  expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
```

Comment:
Maybe testing confirmation modal could be moved to own test?

Reply:
Do you mean in this file? I agree, maybe that would be better